static/src/javascripts/projects/common/modules/article/space-filler.js

```diff
@@ -2,47 +2,57 @@ define([
     'common/utils/fastdom-promise',
     'Promise',
     'raven',
-    'common/modules/article/spacefinder'
+    'common/modules/article/spacefinder',
+    'common/modules/commercial/dfp/QueueAsync'
```

Comment:
This is a good question, and seemed worth investigating.

I created a simple test 'rig' that I could call from the console -

```javascript
window.test = {};
window.test.count = 0;
window.test.qa = QueueAsync;
window.test.op = function () {
    return new Promise(function (resolve, reject) {
        var collection = [];
        window.test.count++;
        for (var i = 0; i < 999; i++) {
            collection.push({});
        }
        window.setTimeout(resolve, 0);
    });
};
window.q = new QueueAsync();
window.test.lotsOps = function () {
    var operations = [];
    for (var i = 0; i < 4999; i++) {
        operations.push(window.q.add(window.test.op))
    }
    return Promise.all(operations);
};
```

Then, in Chrome, I

1. Started recording for a few seconds;
2. Ran `window.test.lotsOps()`;
3. Waited for a console.log set to fire when the lotsOps promise resolved;
4. Waited another few seconds;
5. Triggered a hard GC;
6. Recorded a few seconds afterwards for good measure

...And then looked at my memory allocation graph:

![image](https://cloud.githubusercontent.com/assets/3148617/15319024/b5b5206e-1c20-11e6-9198-8fe2aae1deb3.png)

You can see the heap being thrashed as `test.op` generates its large arrays and incremental GC reclaims the memory. When `lotsOps` completes, there are two sets of 'steps', before I manually trigger garbage collection and heap usage plummets to levels comparable to the start of the run.

On this basis, it seems that if there is memory leakage in our Promise shim, then it doesn't seem to be substantial enough to at least notice on this kind of scale. QueueAsync doesn't seem to leak references to its historical promises, or to the closures around those promises.